ang/civicase/Dashboard.js

```diff
@@ -0,0 +1,207 @@
+        $scope.dashboardActivities.recentCommunication = _.each(data[apiCalls.length - 4].values, formatActivity);
+        $scope.dashboardActivities.recentCommunicationCount = data[apiCalls.length - 3];
+        $scope.dashboardActivities.nextMilestones = _.each(data[apiCalls.length - 2].values, formatActivity);
+        $scope.dashboardActivities.nextMilestonesCount = data[apiCalls.length - 1];
+      });
+    };
+
+    /**
+     * Bind route paramaters to scope variables
+     */
+    function bindRouteParamsToScope () {
+      $scope.$bindToRoute({ param: 'dtab', expr: 'activeTab', format: 'int', default: 0 });
+      $scope.$bindToRoute({ param: 'dbd', expr: 'showBreakdown', format: 'bool', default: false });
+      $scope.$bindToRoute({ param: 'drel', expr: 'filters.caseRelationshipType', format: 'raw', default: 'is_involved' });
+    }
+
+    /**
+     * Watcher for caseRelationshipType
+     *
+     * @param {String} newValue
+     */
+    function caseRelationshipTypeWatcher (newValue) {
+      newValue === 'is_case_manager'
+        ? $scope.activityFilters.case_filter.case_manager = [CRM.config.user_contact_id]
+        : delete ($scope.activityFilters.case_filter.case_manager);
+
+      newValue === 'is_involved'
+        ? $scope.activityFilters.case_filter.contact_id = [CRM.config.user_contact_id]
+        : delete ($scope.activityFilters.case_filter.contact_id);
+    }
+
+    /**
+     * Prepate case filter options
+     */
+    function prepareCaseFilterOption () {
+      var options = [
+        { 'text': 'My cases', 'id': 'is_case_manager' },
+        { 'text': 'Cases I am involved in', 'id': 'is_involved' }
+      ];
+
+      if (CRM.checkPerm('access all cases and activities')) {
+        options.push({ 'text': 'All Cases', 'id': 'all' });
+      }
+
+      $scope.caseRelationshipOptions = options;
+    }
+
+    /**
+     * Initialise watchers
+     */
+    function initWatchers () {
+      $scope.$watch('filters.caseRelationshipType', caseRelationshipTypeWatcher);
+    }
+
+    // Translate between the dashboard's global filter-options and
+    // the narrower, per-section filter-options.
+
+    // $scope.$watch('myCasesOnly', function (myCasesOnly) {
+    //   $scope.activityFilters = {
+    //     case_filter: {'case_type_id.is_active': 1, contact_is_deleted: 0}
+    //   };
+    //   var recentCaseFilter = {
+    //     'status_id.grouping': 'Opened'
+    //   };
+    //   if (myCasesOnly) {
+    //     $scope.activityFilters.case_filter.case_manager = CRM.config.user_contact_id;
+    //     recentCaseFilter.case_manager = [CRM.config.user_contact_id];
+    //   }
+    //   $scope.recentCaseFilter = recentCaseFilter;
+    //   $scope.recentCaseLink = '#/case/list?sf=modified_date&sd=DESC' + (myCasesOnly ? ('&cf=' + JSON.stringify({case_manager: [CRM.config.user_contact_id]})) : '');
+    //   $scope.refresh();
+    // });
+  }
+
+  module.directive('civicaseDashboardTabsetAffix', function ($timeout) {
```

Comment:
Shouldn't we have a different file for it as we are following one directive per file (I think)? Please correct me if I am wrong.

Reply:
For affix directives, we are actually keeping in the same file as the related file. This is also followed in other activity directives. 
Its not ideal I know, but we dont have a proper folder structure also. So lets keep this for now. Once the release is done, we will need to figure out a proper folder structure and place things correctly.